provider/datasource_sites.go

```diff
@@ -0,0 +1,213 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/signalsciences/go-sigsci"
+)
+
+func dataSourceSites() *schema.Resource {
+	return &schema.Resource{
+		Read: readSites,
+		Schema: map[string]*schema.Schema{
+			"filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Filter listed domains by either the site 'name' or 'display_name'",
+			},
+			"sites": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "A list of all sites for a given corp.",
+				Elem: &schema.Resource{
+					// NOTE: The API returns multiple objects with a single 'uri' field.
+					// To avoid extra type complexity we flatten those hierarchies.
+					// These are the fields below that have the '_uri' suffix.
+					Schema: map[string]*schema.Schema{
+						"agent_anon_mode": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Agent IP anonimization mode - 'EU' or 'off'",
+						},
+						"agent_level": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Agent action level - 'block', 'log' or 'off'",
+						},
+						"agents_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's agents",
+						},
+						"alerts_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's alerts",
+						},
+						"analytics_events_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's analytics events",
+						},
+						"blacklist_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's blacklist",
+						},
```

Comment:
**NOTE:** The API has a `blocklist` field but it's not implemented in the `go-sigsci` API client.